### PR TITLE
docs: update and remove some bad content

### PR DESCRIPTION
**Pull request by postamar**
I did a pass on our documentation to check that there wasn't anything in there that could embarrass us by being particularly out of date or misleading.

### CODE REVIEW

File: docs/contributing-to-airbyte/resources/developing-locally.md [hunk 1]
```diff
@@ -1,306 +0,0 @@
...
-1. [Fork](https://docs.github.com/en/github/getting-started-with-github/fork-a-repo) the [`airbyte`](https://github.com/airbytehq/airbyte) repository to develop connectors or the [ `airbyte-platform`](https://github.com/airbytehq/airbyte-platform) repository to develop the Airbyte platform. 
-2. Clone the fork on your workstation:
-
-If developing connectors, you can work on connectors locally but additionally start the platform independently locally using :
-
-   ```bash
-   git clone git@github.com:{YOUR_USERNAME}/airbyte.git
-   cd airbyte
-   ./run-ab-platform.sh
-   ```
```
Comment (evantahler) [inline docs/contributing-to-airbyte/resources/developing-locally.md:28]: This is accurate

File: docs/contributing-to-airbyte/resources/developing-locally.md [hunk 2]
```diff
@@ -1,306 +0,0 @@
...
-If developing platform:
-
-   ```bash
-   git clone git@github.com:{YOUR_USERNAME}/airbyte-platform.git
-   cd airbyte-platform
-   docker compose up
-   ```
```
Comment (evantahler) [inline docs/contributing-to-airbyte/resources/developing-locally.md:35]: this is also accurate

File: docs/contributing-to-airbyte/resources/developing-locally.md
Comment (evantahler) [inline docs/contributing-to-airbyte/resources/developing-locally.md:1]: This page is confusing, but I think partially correct and we should keep something like this around.

File: docs/contributing-to-airbyte/resources/developing-locally.md [hunk 3]
```diff
@@ -1,306 +0,0 @@
...
-
-Note: If you are contributing a Python file without imports or function definitions, place the following comment at the top of your file:
-
-```python
-"""
-[FILENAME] includes [INSERT DESCRIPTION OF CONTENTS HERE]
-"""
-```
-
-### Develop on `airbyte-webapp`
```
Comment (evantahler) [inline docs/contributing-to-airbyte/resources/developing-locally.md:189]: This is all still correct

File: docs/contributing-to-airbyte/resources/developing-on-docker.md
Comment (evantahler) [inline docs/contributing-to-airbyte/resources/developing-on-docker.md:1]: Check with the platform folks on this - it might still be valid

**Review by evantahler — CHANGES_REQUESTED** [review]

File: docs/contributing-to-airbyte/resources/developing-on-docker.md
Comment (postamar) [reply docs/contributing-to-airbyte/resources/developing-on-docker.md:1]: Oh, this is for the platform. My bad.

File: docs/contributing-to-airbyte/resources/developing-locally.md
Comment (postamar) [reply docs/contributing-to-airbyte/resources/developing-locally.md:1]: I'll see what I can to. I think I might have gotten confused.

**Review by evantahler — APPROVED** [review]

**Pull request merged**